### ISSUE TITLE
Cluster deploys fail with SGX node pools when clusterSubnet configured

### ISSUE THREAD

**Issue by artburkart** [Azure/aks-engine#2329]
**Describe the bug**
If I take the `apimodel` key from [sgx.json](https://github.com/Azure/aks-engine/blob/0f99cf20fc1e985b3b8c43c084aa077e5bd55e03/test/e2e/test_cluster_configs/sgx.json) and run it as-is, I can successfully deploy a cluster with an SGX node pool. However, if I specify a value for `properties.orchestratorProfile.kubernetesConfig.clusterSubnet`, then it fails.

Here's the error message I get:
```json
OS Provisioning for VM 'k8s-sgx0-23495343-vmss_0' did not finish in the allotted time. The VM may still finish provisioning successfully. Please check provisioning state later. For details on how to check current provisioning state of Windows VMs, refer to https://aka.ms/WindowsVMLifecycle and Linux VMs, refer to https://aka.ms/LinuxVMLifecycle.
```

**Steps To Reproduce**
```diff
{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
+     "kubernetesConfig": {
+       "clusterSubnet": "10.135.0.0/16"
+     }
    },
    "masterProfile": {
      "count": 1,
      "dnsPrefix": "test-sgx",
      "vmSize": "Standard_DS2_v2",
      "OSDiskSizeGB": 200,
      "availabilityProfile": "AvailabilitySet"
    },
    "agentPoolProfiles": [
      {
        "name": "sgx0",
        "count": 1,
        "vmSize": "Standard_DC2s",
        "OSDiskSizeGB": 200,
        "storageProfile": "ManagedDisks",
        "availabilityProfile": "VirtualMachineScaleSets"
      }
    ],
    "linuxProfile": {
      "adminUsername": "azureuser",
      "ssh": {
        "publicKeys": [
          {
            "keyData": ""
          }
        ]
      }
    },
    "servicePrincipalProfile": {
      "clientId": "",
      "secret": ""
    }
  }
}
```

**Expected behavior**
The SGX node pool should successfully deploy. 

Clusters successfully deploy when I change the VM type from `Standard_DC2s` to anything else such as `Standard_DS2_v2`.

**AKS Engine version**
```
Version: v0.43.0
GitCommit: 8928a4094
GitTreeState: clean
```

**Kubernetes version**
1.13.12

**Comment by jackfrancis** [comment]
Hi @artburkart,

Sorry if this is scantly documented, but an explicit `clusterSubnet` configuration is designed to pair w/ a custom VNET scenario. Simply including _only_ a custom `clusterSubnet` value doesn't make sense, because:

- on the one hand, you're trusting AKS Engine to create a VNET for you
- on the other hand, you're relying on your custom subnet CIDR to be a CIDR that makes sense in the context of the VNET that's out of your control

See here for the description of the aks-engine `clusterSubnet` config property:

https://github.com/Azure/aks-engine/blob/master/docs/topics/clusterdefinitions.md#kubernetesconfig

tl;dr to use the `customSubnet` configuration you should also pre-create a VNET with the right CIDR settings to accommodate your Kubernetes cluster

See:

https://github.com/Azure/aks-engine/blob/master/docs/topics/clusterdefinitions.md#kubernetesconfig

**Comment by artburkart** [comment]
Thanks @jackfrancis. 

I actually did find that in the docs and configured accoro. Unfortunately, I observe the same behavior whether I configure it with the vnetSubnetId or not. 

However, in both cases, I'm able to successfully deploy a cluster if I use different VM sizes. Would it be helpful if I shared my VNet configuration? If so, in what format should I share it?

Best,
Arthur

**Comment by artburkart** [comment]
I'm going to double-check my work; will get back to you.

**Comment by artburkart** [comment]
My apologies. I think I was getting hung up on the fact that `aks-engine` assumes my VNet only has one entry in `addressPrefixes`. Once I stopped trying to deploy my VMs to a secondary address prefix, things started to work. Although, I'm not sure that entirely accounts for the discrepancy between deploying SGX VMs and other VM sizes. 

In any case, I'll close this issue. Thanks again for your help.

**Comment by jackfrancis** [comment]
Please don't hesitate to bug us again, perhaps there are some documentation gaps; and you're right, the VM SKU shouldn't matter in terms of these networking configs
